tensorflow/python/layers/base.py

```diff
@@ -354,3 +355,13 @@ def _add_elements_to_collection(elements, collection_list):
       if element not in collection_set:
         collection.append(element)
 
+def _should_add_regularizer(variable, existing_variable_set):
+  result = True
+  if isinstance(variable, tf_variables.PartitionedVariable):
+    for var in variable._get_variable_list():
```

Comment:
variable is an iterable, so you should be able to simply do `for var in variable:` (see https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/ops/variables.py#L1295).